src/qibolab/pulses/sequence.py

```diff
@@ -2,158 +2,45 @@
 
 from collections import defaultdict
 
-from .pulse import PulseType
+from .pulse import Delay, PulseType
 
 
-class PulseSequence(list):
-    """A collection of scheduled pulses.
+class PulseSequence(defaultdict):
```

Comment:
I'd be in favor of a renaming `PulseSequence` -> `Sequence` (since there is no other meaningful sequence for the Qibolab user, and internally will be scoped in `pulses` anyhow). But it's definitely not required.

Would you agree with that? @stavros11 @hay-k

Reply:
Yes, that should be done. I can do another followup PR just for the rename, since it is going to cause a lot changes.